src/Author.php

```diff
@@ -33,13 +28,26 @@ class Author extends AbstractEndpoint {
 	 * @return array The array with the data of the endpoint
 	 */
 	public function endpoint_callback( \WP_REST_Request $request ) {
-		$user_id = $request->get_param( 'id' );
-		$response = [];
-		$user = get_user_by( 'id', $user_id );
+		$params = $request->get_params();
+
+		$id = ( false === $params['id'] ) ? false : absint( $params['id'] );
+
+		$slug = ( false === $params['slug'] ) ? false : trim( $params['slug'], '/' );
+
+		if ( false === $id && false === $slug ) {
+			return new \WP_Error( self::INVALID_PARAMS, 'The request must have either an id or a slug', [ 'status' => 400 ] );
```

Comment:
I guess its personal preference, there is not much I can say about that.

However, I understand that we will receive only 1 parameter, so one of these 2 lines is always going to throw a warning:

```
$id =  absint( $params['id'] );
$slug = trim( sanitize_text_field ( $params['slug'], '/' ) );
```

On the other hand, it is arguable which variation is simpler.